flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/api/EnvironmentSettings.java

```diff
@@ -59,9 +62,17 @@ public class EnvironmentSettings {
 
     private final ClassLoader classLoader;
 
+    private final @Nullable CatalogStore catalogStore;
```

Comment:
Why do we need to change `EnvironmentSettings`? I think we should create a catalog storage by the configuration instead of maintaining an instance here.

Reply:
We need to implement a method for users to register the catalogStore. This behavior is part of the FLIP design.